### DISCUSSION TITLE
Sort methods based on <?>

### DISCUSSION THREAD

**Discussion by Kludex** [astral-sh/ruff#5382] [Ideas]
Hi 👋 

This is just an idea, not a feature request (but if you like, it is).

On Pydantic, there was a PR reordering the methods on a class (https://github.com/pydantic/pydantic/pull/6260), and there was an internal discussion about "how to enforce" this kind of change. And well... I thought about suggesting two rules:

- DM001: Sort methods in "logical order" - which would be similar to what you see on that PR (`__init__`, then non dunder methods, then dunder methods).
- DM001: Sort methods alphabetically.

Do you think this makes sense?

**Comment by zanieb** [comment]
I believe this would be covered by https://github.com/astral-sh/ruff/issues/3946 / #5271 — seems really nice to have!

**Reply by Kludex** [reply]
Are you following me @zanieb ? 🤣 

(Thanks! 🙏)
